src/Microsoft.Health.Fhir.Shared.Core.UnitTests/Features/Resources/Patch/PatchResourceHandlerTests.cs

```diff
@@ -82,4 +82,138 @@ public async Task GivenAPatchResourceHandler_WhenHandlingAPatchResourceRequestWi
+
+        authService
+            .CheckAccess(DataActions.Update | DataActions.Read | DataActions.Write, CancellationToken.None)
+            .Returns(DataActions.Update);
+
+        ResourceElement patient = Samples.GetDefaultPatient().UpdateVersion("1");
+        var wrapper = new ResourceWrapper(
+            patient,
+            new RawResource(patient.Instance.ToJson(), FhirResourceFormat.Json, false),
+            new ResourceRequest(HttpMethod.Get),
+            false,
+            null,
+            null,
+            null);
+
+        fhirDataStore.GetAsync(Arg.Any<ResourceKey>(), Arg.Any<CancellationToken>())
+            .Returns(Task.FromResult(wrapper));
+
+        var request = new PatchResourceRequest(new ResourceKey("Patient", "123"), new FhirPathPatchPayload(new Parameters()), bundleResourceContext: null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await patchHandler.Handle(request, CancellationToken.None);
+
+        await mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<UpsertResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAPatchResourceHandler_WhenUserHasLegacyReadAndWritePermissions_ThenPatchShouldSucceed()
+    {
+        // Arrange
+        IAuthorizationService<DataActions> authService = Substitute.For<IAuthorizationService<DataActions>>();
+        IFhirDataStore fhirDataStore = Substitute.For<IFhirDataStore>();
+        IMediator mediator = Substitute.For<IMediator>();
+
+        var patchHandler = Mock.TypeWithArguments<PatchResourceHandler>(mediator, authService, fhirDataStore);
+
+        authService
+            .CheckAccess(DataActions.Update | DataActions.Read | DataActions.Write, CancellationToken.None)
+            .Returns(DataActions.Read | DataActions.Write);
+
+        ResourceElement patient = Samples.GetDefaultPatient().UpdateVersion("1");
+        var wrapper = new ResourceWrapper(
+            patient,
+            new RawResource(patient.Instance.ToJson(), FhirResourceFormat.Json, false),
+            new ResourceRequest(HttpMethod.Get),
+            false,
+            null,
+            null,
+            null);
+
+        fhirDataStore.GetAsync(Arg.Any<ResourceKey>(), Arg.Any<CancellationToken>())
+            .Returns(Task.FromResult(wrapper));
+
+        var request = new PatchResourceRequest(new ResourceKey("Patient", "123"), new FhirPathPatchPayload(new Parameters()), bundleResourceContext: null);
+
+        // Act & Assert - Should not throw UnauthorizedFhirActionException
+        await patchHandler.Handle(request, CancellationToken.None);
+
+        await mediator
+            .Received()
+            .Send<UpsertResourceResponse>(Arg.Any<UpsertResourceRequest>(), Arg.Any<CancellationToken>());
+    }
+
+    [Fact]
+    public async Task GivenAPatchResourceHandler_WhenUserLacksUpdatePermission_ThenUnauthorizedExceptionIsThrown()
```

Comment:
Just a suggestion - should we club all the UnauthorizedFhirActionException facts together into a theory?

Reply:
Theory created.